Source/Posts/2016/08-Performance-boost-for-Hoa-Compiler.xyl

```diff
@@ -0,0 +1,185 @@
+<?xml version="1.0" encoding="utf-8"?>
+<?xyl-overlay href="hoa://Application/Overlays/Article.xyl"?>
+<?xyl-meta name="title" value="Performance boost for <code>Hoa\Compiler</code>"?>
+<?xyl-meta name="date" value="2016-08-22T08:17:28+02:00"?>
+
+<overlay xmlns="http://hoa-project.net/xyl/xylophone">
+  <article id="main">
+    <p>
+      A lot of work has been done <strong>to boost the performance</strong> of
+      <a href="http://central.hoa-project.net/Resource/Library/Compiler"><code>Hoa\Compiler</code></a>,
+      mainly by reducing the amount of memory required to lexically and to
+      syntactically analyse a data. <strong>Up to 80% of the
+      memory can be saved</strong> with a large data. Also a new feature allows
+      to compile the in-memory parser into a PHP class. <strong>Up to 45% of the
+      CPU can be saved</strong> for the parser initialisation.
+    </p>
+
+    <h2>Constant memory usage in the lexer</h2>
+
+    <p>
+      First of all, one of the biggest improvement lands in the lexer (aka the
+      lexical analyser). The lexer is responsible of cutting a datum into a
+      sequence of tokens (aka lexemes). So it reads the entire datum and
+      produces an entire sequence.
+      (<a href="http://hoa-project.net/Literature/Hack/Compiler.html">Learn more
+      by reading its documentation</a>). Until today.
+    </p>
+    <p>
+      The issue with this approach is that if the datum is of length 1Mb, then
+      the memory will contain 1Mb (the original datum) and at least 1Mb (the
+      sequence). However, the sequence is represented as a PHP array, so it has
+      an overhead of some bytes per tokens, in addition to the meta data attached
+      to the token (like the name, the namespace, the length, the offset etc.).
+      The resulting overhead size will depend of the PHP version in use but this
+      is not optimal.
+    </p>
+    <p>
+      Thus, a straightforward optimisation consists at transforming the lexer into
+      an iterator. Actually, it has been <strong>transformed into a
+      generator</strong>. So each call to the lexer will produce the next
+      token, without stacking all the tokens in memory.
+    </p>
+    <p>
+      With this new approach, the API stays <strong>identically the
```

Comment:
"identically the same" seems repetitive. Why not just use either "identical" or "the same" ?